REVIEWING.md

```diff
@@ -0,0 +1,84 @@
+# Reviewing pull requests
+
+This document provides a high-level and **incomplete** checklist of things that pull request reviewers should be aware of when performing code review. The checklist mainly consists of things that are specific to the scikit-bio project and that generally apply to incoming pull requests. The checklist is incomplete because it is not possible to describe all things to verify during code review (that depends on what is being reviewed). This document also doesn't attempt to describe *how* to perform code review (there are many online resources for that).
+
+Reviewers are encouraged to keep this document up-to-date as the project evolves and to add anything that's missing.
+
+The checklist is not in any particular order.
+
+## Changelog
+
+This is one of the easiest parts to forget in a pull request.
```

Comment:
"...in a pull request, but is one of the most important points to remember as the changelog is the document which developers will review to identify changes relevant to them."